lib/modelscan_api/bailo_modelscan_api/main.py

```diff
@@ -101,6 +130,7 @@ def scan_file(in_file: UploadFile, background_tasks: BackgroundTasks) -> dict[st
     finally:
         try:
             # Clean up the downloaded file as a background task to allow returning sooner.
+            # If using a temporary dir then this would happen anyway, but if Settings' download_dir evaluates then this is required.
             logger.info("Cleaning up downloaded file.")
             background_tasks.add_task(Path.unlink, pathlib_path, missing_ok=True)
         except UnboundLocalError:
```

Comment:
When logging the exception could you include info about which file has caused the error?

Reply:
`logger.exception` automatically includes the raised exception details